app/models/internal/impl/DefaultReport.java

```diff
@@ -53,6 +55,11 @@ public UUID getId() {
         return _id;
     }
 
+    @Override
+    public String getETag() {
+        return DigestUtils.sha1Hex(toString());
```

Comment:
Really shouldn't use `toString()` to compute etag; this should probably come from the repository/data model. Feel free to leave it for now with a `TODO`. @cwbriones let me know if you think otherwise.